src/DSharp.Compiler/ScriptModel/Symbols/SymbolSet.cs

```diff
@@ -900,10 +900,13 @@ public TypeSymbol ResolveType(ParseNode node, ISymbolTable symbolTable, Symbol c
                     typeArguments.Add(argType);
                 }
 
-                TypeSymbol resolvedSymbol = CreateGenericTypeSymbol(templateType, typeArguments);
-                Debug.Assert(resolvedSymbol != null);
+                if(templateType != null)
+                {
+                    TypeSymbol resolvedSymbol = CreateGenericTypeSymbol(templateType, typeArguments);
+                    Debug.Assert(resolvedSymbol != null);
```

Comment:
Debug.Assert doesn't give any warning on normal builds

Reply:
That's existing code, i'm not that bothered about reworking it right now.